docs/DEV.md

```diff
@@ -0,0 +1,179 @@
+# Developing this add-on
+
+### Preparations
+
+* Download a Developer and Nightly versions of Firefox (only Developer/Nightly will allow running unsigned legacy extensions, and Nightly is the default target for the automated tests)
+
+## Getting started
+
+```bash
+# install dependencies
+npm install
+
+## build
+npm run eslint
+npm run build
+
+## build and run
+npm run firefox
+```
+
+## Details
+
+First, make sure you are on NPM 5+ installed so that the proper dependencies are installed using the package-lock.json file.
+
+`$ npm install -g npm`
+
+Clone the repo:
+
+`$ git clone https://github.com/mozilla/shield-studies-addon-template.git`
+
+After cloning the repo, you can run the following commands from the top level directory, one after another:
+
+```
+$ npm install
+$ npm run build
+```
+
+This packages the add-on into an xpi file which is stored in `dist/`. This file is what you load into Firefox.
+
+## General notes on Shield Study Engineering
+
+Shield study add-ons are legacy (`addon/bootstrap.js`) add-ons with an optional embedded web extension (`addon/webextension/background.js`).
+
+The web extension needs to be packaged together with a legacy add-on in order to be able to access Telemetry data, user preferences etc that are required for collecting relevant data for [Shield Studies](https://wiki.mozilla.org/Firefox/Shield/Shield_Studies).
+
+It is recommended to build necessary logic and user interface using in the context of the webextension and communicate with the legacy add-on code through messaging whenever privileged access is required.
+
+## Loading the Web Extension in Firefox
+
+You can have Firefox automatically launched and the add-on installed by running:
+
+`$ npm run firefox`
+
+To load the extension manually instead, open (preferably) the [Developer Edition of Firefox](https://www.mozilla.org/firefox/developer/) and load the `.xpi` using the following steps:
+
+* Navigate to _about:config_ and set `extensions.legacy.enabled` to `true`. This permits the loading of the embedded Web Extension since new versions of Firefox are becoming restricted to pure Web Extensions only.
+* Navigate to _about:debugging_ in your URL bar
+* Select "Load Temporary Add-on"
+* Find and select the latest xpi file you just built.
+
+## Seeing the add-on in action
+
+To debug installation and loading of the add-on:
+
+* Navigate to _about:config_ and set `shield.testing.logging.level` to `10`. This permits shield-add-on log output in browser console
```

Comment:
This currently doesn't do anything right now, so we should probably remove it until SSAU (utils) gets a log?